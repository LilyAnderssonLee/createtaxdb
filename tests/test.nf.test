nextflow_pipeline {

    name "Test pipeline: NFCORE_CREATETAXDB"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_createtaxdb"
    tag "test"

    test("test_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                        path("$outputDir/diamond/database.dmnd"),
                        path("$outputDir/kaiju/database.fmi"),
                        path("$outputDir/malt/malt-build.log").readLines().last().contains('Peak memory'),
                        path("$outputDir/malt/malt_index/index0.idx"),
                        path("$outputDir/malt/malt_index/ref.db"),
                        path("$outputDir/malt/malt_index/ref.idx"),
                        path("$outputDir/malt/malt_index/ref.inf"),
                        path("$outputDir/malt/malt_index/taxonomy.idx"),
                        path("$outputDir/malt/malt_index/taxonomy.map"),
                        path("$outputDir/malt/malt_index/taxonomy.tre")
                    ).match()
                },
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert path("$outputDir/malt/malt_index/table0.db").exists() },
                { assert path("$outputDir/malt/malt_index/table0.idx").exists() },
            )
        }
    }
}
