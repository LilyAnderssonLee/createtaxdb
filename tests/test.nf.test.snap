{
    "test_profile": {
        "content": [
            "database.dmnd:md5,9d57aa88cd1766adfda8360876fc0e4f",
            "database.fmi:md5,54fd89f5e4eab61af30175e8aa389598",
            "software_versions.yml:md5,e807fbfb5969b28495024decc5a639a1"
        ],
        "timestamp": "2023-12-14T12:29:56.030615203"
    }
}